hackathon_site/event/models.py

```diff
@@ -0,0 +1,39 @@
+from django.db import models
+from django.contrib.auth.models import User
+
+import uuid
+
+
+def _generate_team_code():
+    team_code = uuid.uuid4().hex[:5].upper()
+    while Team.objects.filter(team_code=team_code).exists():
+        team_code = uuid.uuid4().hex[:5].upper()
+    return team_code
+
+
+class Team(models.Model):
+    team_code = models.CharField(max_length=5, default=_generate_team_code, null=False)
+
+    created_at = models.DateTimeField(auto_now_add=True, null=False)
+    updated_at = models.DateTimeField(auto_now=True, null=False)
+
+    MAX_MEMBERS = 4
+
+    def __str__(self):
+        return self.team_code
+
+
+class Profile(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    team = models.ForeignKey(Team, on_delete=models.CASCADE, null=False)
+    status = models.CharField(max_length=64, default=None, null=True)
```

Comment:
Could we specify the choices for this now? We can add them in later if we're not sure at this point